source/CIPauseMenu.cpp

```diff
@@ -0,0 +1,151 @@
+//
+//  CIPauseMenu.cpp
+//  CoreImpact
+//
+//  Created by Richard Yoon on 4/5/21.
+//  Copyright � 2021 Game Design Initiative at Cornell. All rights reserved.
+//
+
+#include "CIPauseMenu.h"
+
+using namespace cugl;
+
+#pragma mark -
+#pragma mark Constructors
+
+/**
+ * Disposes of all (non-static) resources allocated to this menu.
+ */
+void PauseMenu::dispose() {
+    _settingsTitle = nullptr;
+    _musicLabel = nullptr;
+    _volumeLabel = nullptr;
+    _parallaxLabel = nullptr;
+
+    _musicBtn = nullptr;
+    _volumeSlider = nullptr;
+    _parallaxBtn = nullptr;
+
+    _resumeBtn = nullptr;
+    _exitBtn = nullptr;
+
+    _layer = nullptr;
+}
+
+/**
+ * Initializes a new pause menu with the state pointer.
+ *
+ * @param assets        The (loaded) assets for this pause menu
+ *
+ * @return true if initialization was successful, false otherwise
+ */
+bool PauseMenu::init(const std::shared_ptr<cugl::AssetManager>& assets,
+    const std::shared_ptr<NetworkMessageManager>& networkMessageManager, 
+    const std::shared_ptr<PlayerSettings>& playerSettings) {
+    // Initialize the scene to a locked width
+    Size dimen = Application::get()->getDisplaySize();
+    // Lock the scene to a reasonable resolution
+    dimen *= CONSTANTS::SCENE_WIDTH / dimen.width;
+    if (assets == nullptr) {
+        return false;
+    }
+
+    _layer = assets->get<scene2::SceneNode>("pause");
+    _layer->setContentSize(dimen);
+    _layer->doLayout();
+
+    _playerSettings = playerSettings;
+
+    _settingsTitle = std::dynamic_pointer_cast<scene2::Label>(assets->get<scene2::SceneNode>("pause_title"));
+    _musicLabel = std::dynamic_pointer_cast<scene2::Label>(assets->get<scene2::SceneNode>("pause_musiclabel"));
+    _volumeLabel = std::dynamic_pointer_cast<scene2::Label>(assets->get<scene2::SceneNode>("pause_volumelabel"));
+    _parallaxLabel = std::dynamic_pointer_cast<scene2::Label>(assets->get<scene2::SceneNode>("pause_parallaxlabel"));
```

Comment:
do we want the parallax label? I don't think we need it in the pause menu

Reply:
its in the designs. @JackieW001?

Reply:
My thoughts were that we would be able to have parallax effect, but if we aren't actually going to implement it, we should take it out.